packages/core/strapi/lib/core/app-configuration/index.js

```diff
@@ -80,20 +81,16 @@ module.exports = (dir, initialConfig = {}) => {
       ...pkgJSON,
       strapi: strapiVersion,
     },
-    installedPlugins: getPrefixedDeps('@strapi/plugin', pkgJSON),
+    // installedPlugins: getPrefixedDeps('@strapi/plugin', pkgJSON),
     installedMiddlewares: getPrefixedDeps('@strapi/middleware', pkgJSON),
     installedHooks: getPrefixedDeps('@strapi/hook', pkgJSON),
-    installedProviders: getPrefixedDeps('@strapi/provider', pkgJSON),
+    // installedProviders: getPrefixedDeps('@strapi/provider', pkgJSON),
   };
 
-  const baseConfig = {
-    ...loadConfigDir(configDir),
-    policies: loadPolicies(path.resolve(configDir, 'policies')),
-    functions: loadFunctions(path.resolve(configDir, 'functions')),
-  };
+  const baseConfig = omit('plugins', loadConfigDir(configDir)); // plugin config will be loaded later
 
   const envDir = path.resolve(configDir, 'env', process.env.NODE_ENV);
   const envConfig = loadConfigDir(envDir);
 
-  return createConfigProvider(_.merge(rootConfig, defaultConfig, baseConfig, envConfig));
+  return _.merge(rootConfig, defaultConfig, baseConfig, envConfig);
```

Comment:
feels like not keep createConfigProvider make the usage in the commands inconsistent. we might want to rethink this

Reply:
Yes we can rethink it, after merging?